app/services/middleware_topology_service.rb

```diff
@@ -11,11 +11,11 @@ class MiddlewareTopologyService < TopologyService
       :middleware_deployments,
       :middleware_datasources,
       :middleware_messagings,
-      :lives_on => [:host]
+      :lives_on => [:host, :container]
```

Comment:
yes, the linking in the parser is correct, and it works here due to the fallback that you added.
but right now the hierarchy as defined in topology above is:` eap.lives_on.container` (while lives_on is the container itself) - it doesn't fail due to the fallback check, but it doesn't seem to be correct either.